lib/krikri/util/extended_date_parser.rb

```diff
@@ -55,6 +56,22 @@ def range_match(str)
       end
     end
 
+    def preprocess(str)
+      str.gsub!(/late/i, '')
+      str.strip!
+      str.gsub!(/\s+/, ' ')
+      str.gsub!('-', 'x') if str.match(/^[1-9]+\-+$/)
+      str.gsub!('0s', 'x') if str.match(/^[1-9]+0s$/)
```

Comment:
Are these characters being turned into "x" so that they can be matched more easily in the other functions?

If so, could comments be added wherever this applies?